docs/en_US/Tutorial/InstallCustomizedAlgos.md

```diff
@@ -0,0 +1,155 @@
+The `ClassArgsValidator` interface is defined as:
+```python
+class ClassArgsValidator(object):
+    def validate_class_args(self, **kwargs):
+        """
+        The classArgs fields in experiment configuration are packed as a dict and
+        passed to validator as kwargs.
+        """
+        pass
+```
+For example, you can implement your validator such as:
+```python
+from schema import Schema, Optional
+from nni import ClassArgsValidator
+
+class MedianstopClassArgsValidator(ClassArgsValidator):
+    def validate_class_args(self, **kwargs):
+        Schema({
+            Optional('optimize_mode'): self.choices('optimize_mode', 'maximize', 'minimize'),
+            Optional('start_step'): self.range('start_step', int, 0, 9999),
+        }).validate(kwargs)
+```
+The validator will be invoked before experiment is started to check whether the classArgs fields are valid for your customized algorithms.
+
+### 3. Prepare package installation source
+In order to be installed as builtin tuners, assessors and advisors, the customized algorithms need to be packaged as installable source which can be recognized by `pip` command, under the hood nni calls `pip` command to install the package.
+Besides being a common pip source, the package needs to provide meta information in the `classifiers` field.
+Format of classifiers field is a following:
+```
+NNI Package :: <type> :: <builtin name> :: <full class name of tuner> :: <full class name of class args validator>
+```
+* `type`: type of algorithms, could be one of `tuner`, `assessor`, `advisor`
+* `builtin name`: builtin name used in experiment configuration file
+* `full class name of tuner`: tuner class name, including its module name, for example: `demo_tuner.DemoTuner`
+* `full class name of class args validator`: class args validator class name, including its module name, for example: `demo_tuner.MyClassArgsValidator`
+Following is an example of classfiers in package's `setup.py`:
+
+```python
+    classifiers = [
+        'Programming Language :: Python :: 3',
+        'License :: OSI Approved :: MIT License',
+        'Operating System :: ',
+        'NNI Package :: tuner :: demotuner :: demo_tuner.DemoTuner :: demo_tuner.MyClassArgsValidator'
+    ],
+```
+
+Once you have the meta info in `setup.py`, you can build your pip installation source via:
+* Run command `python setup.py develop` from the package directory, this command will build the directory as a pip installation source.
+* Run command `python setup.py bdist_wheel` from the package directory, this command build a whl file which is a pip installation source.
+
+NNI will look for the classifier starts with `NNI Package` to retrieve the package meta information while the package being installed with `nnictl package install <source>` command.
+
+Reference [customized tuner example](https://github.com/microsoft/nni/blob/master/examples/tuners/customized_tuner/README.md) for a full example.
```

Comment:
this is a broken link

Reply:
It links to a README.md not yet checked in.  The link will be working once PR is merged.